objects/src/block/block_number.rs

```diff
@@ -22,9 +22,12 @@ impl BlockNumber {
     /// exponent.
     pub const EPOCH_LENGTH_EXPONENT: u8 = 16;
 
+    /// Genesis BlockNumber
+    pub const GENESIS: Self = Self(0);
+
     /// Returns the previous block number
-    pub fn parent(&self) -> u32 {
-        self.checked_sub(1).expect("Cannot go below Genesis block!").0
+    pub fn parent(&self) -> Option<u32> {
```

Comment:
These should probably return `Option<BlockNumber>`, same with `fn child`.

Reply:
Should `child` actually return `Option<BlockNumber>`? It seems fine to return `BlockNumber` since a u32 overflow would in practice only occur decades from now.
For parent I think it makes sense because in practice one might actually call `parent()` on the genesis block.